### PR TITLE
Use dep instead of godeps

### LINKED ISSUE #232: Evaluate vendor manage tools
We are using godep for managing vendors today, but it is not so easy to update standalone packages with godep. We should consider an alternative one. Some candidates include:

- [vndr](https://github.com/LK4D4/vndr): containerd and moby are using it.
- [glide](https://github.com/Masterminds/glide): helm and kompose are using it.
- [dep](https://github.com/golang/dep): dashboard is using it. But it's still not stable now, hoping to be the official toolchain in go 1.10.

I'd prefer vndr now, and we should switch to dep when it is stable.

### PR DESCRIPTION
**Pull request by resouer**
This is PR to use dep to replace godep

Fixes #232

I can not successfully use vndr, so I chose dep. Also, vndr is maintained by one person while dep is a promising official tool.

### CODE REVIEW

**Comment by feiskyer** [comment]
@resouer conflicting godeps.json.

**Comment by resouer** [comment]
@feiskyer fixed

File: Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+[[constraint]]
+  name = "google.golang.org/grpc"
+  version = "1.0.4"
+
+[[constraint]]
+  name = "k8s.io/apimachinery"
+
+[[constraint]]
+  name = "k8s.io/client-go"
```
Comment (feiskyer) [inline Gopkg.toml]: Seems missing version for some packages. Add version or revision for all packages?

**Comment by feiskyer** [comment]
could you also update `hack/godep-save.sh` to `hack/update-vendors.sh`?

File: Gopkg.toml (hunk 1 above)
Comment (resouer) [reply Gopkg.toml]: This file is used to specify user requirement for dependencies (for example: fork), see: https://github.com/golang/dep/blob/master/docs/Gopkg.toml.md#constraint, the reversions are enforced by `.lock` file.

 Though I don't know why there're so many pkg been listed in this file.

**Pull request merged**